### PR TITLE
Implement parser

### LINKED ISSUE #56: Integrate parser into Tutorlink class file

### PR DESCRIPTION
**Pull request by yeekian**
Fixes #56

### CODE REVIEW

File: src/main/java/tutorlink/parser/Parser.java [hunk 1]
```diff
@@ -17,15 +25,66 @@ private String extractCommandWord(String input) {
...
+
+        case ListStudentCommand.COMMAND_WORD:
+            return new ListStudentCommand(); // Lists all students
+
+        case ExitCommand.COMMAND_WORD:
+            return new ExitCommand(); // Lists all students
+
+        default:
+            LOGGER.warning(UNKNOWN_COMMAND_ERROR_MESSAGE);
+            return null; // or handle unknown command appropriately
```
Comment (ThienDuc3112) [inline src/main/java/tutorlink/parser/Parser.java]: can just throw invalid command and it'll auto get handle by current main loop

File: src/main/java/tutorlink/parser/Parser.java [hunk 2]
```diff
@@ -17,15 +25,66 @@ private String extractCommandWord(String input) {
...
 
+        // Build regex pattern dynamically from the provided argument prefixes
+        StringBuilder regexBuilder = new StringBuilder("(?i)(");
+        for (String prefix : argumentPrefixes) {
+            regexBuilder.append(Pattern.quote(prefix)).append("|");
+        }
+        regexBuilder.setLength(regexBuilder.length() - 1); // Remove the last "|"
+        regexBuilder.append(")([^\\s]+)"); // Capture the argument after the prefix
+
+        String regex = regexBuilder.toString();
```
Comment (ThienDuc3112) [inline src/main/java/tutorlink/parser/Parser.java:77]: LGTM, this'll work for now

**Review by ThienDuc3112 — APPROVED** [review]
LGTM

File: src/main/java/tutorlink/parser/Parser.java [hunk 3]
```diff
@@ -17,15 +25,66 @@ private String extractCommandWord(String input) {
...
-    }
+    public HashMap<String, String> getArguments(String[] argumentPrefixes, String line) {
+        HashMap<String, String> arguments = new HashMap<>();
 
+        if (argumentPrefixes == null) {
+            return arguments;
+        }
 
+        // Build regex pattern dynamically from the provided argument prefixes
+        StringBuilder regexBuilder = new StringBuilder("(?i)(");
```
Comment (jinzihan2002) [inline src/main/java/tutorlink/parser/Parser.java:70]: I like the use of StringBuilder to dynamically generate the regex pattern

**Review by jinzihan2002 — APPROVED** [review]

File: src/main/java/tutorlink/parser/Parser.java (hunk 3 above)
Comment (jinzihan2002) [inline src/main/java/tutorlink/parser/Parser.java:70]: I like the use of StringBuilder to dynamically generate the regex pattern

**Review by jinzihan2002 — APPROVED** [review]
LGTM

File: src/main/java/tutorlink/parser/Parser.java (hunk 1 above)
Comment (yeekian) [reply src/main/java/tutorlink/parser/Parser.java]: Thank you, I have implemented your suggestion.

**Pull request merged**